tests/unit/UsersControllerTest.php

```diff
@@ -1777,6 +1780,551 @@ public function testCreateSuccessfulWithValidEmailAdmin() {
+
+			$result = $this->createController()->setPasswordForm('fooBaZ1', 'foo');
+			$this->assertEquals(
+				new Http\TemplateResponse('user_management', 'new_user/resendtokenbymail',
+					['link' => 'http://localhost/apps/user_management/setpassword/form/1234/foo'],
+					'guest'), $result);
+		} elseif ($tokenException === UserTokenException::TOKEN_MISMACTH) {
+			$this->config->expects($this->once())
+				->method('getUserValue')
+				->willReturn('1234:fooBaZ11');
+			$this->timeFactory->expects($this->once())
+				->method('getTime')
+				->willReturn(44430);
+			$result = $this->createController()->setPasswordForm('fooBaZ1', 'foo');
+			$this->assertEquals(
+				new Http\TemplateResponse(
+					'core', 'error',
+					['errors' => [['error' => 'The token mismatch happened.']]], 'guest'), $result
+			);
+		} elseif ($tokenException === UserTokenException::TOKEN_AND_TIME_INVALID) {
+			$this->config->expects($this->once())
+				->method('getUserValue')
+				->willReturn('');
+			$result = $this->createController()->setPasswordForm('fooBaZ1', 'foo');
+			$this->assertEquals(
+				new Http\TemplateResponse('core', 'error',
+					['errors' => [["error" => 'The token provided is invalid.']]], 'guest'), $result);
+		}
+	}
+
+	public function testResendToken() {
+		$user = $this->createMock(IUser::class);
+		$user->method('getEMailAddress')
+			->willReturn('foo@bar.com');
+
+		$this->userManager->expects($this->once())
+			->method('get')
+			->willReturn($user);
+
+		$this->secureRandom->expects($this->once())
+			->method('generate')
+			->willReturn('foOBaZ1');
+		$this->urlGenerator->expects($this->once())
+			->method('linkToRouteAbsolute')
+			->willReturn('http://localhost/setpassword/foOBaZ1/foo');
+
+		$message = $this->createMock(Message::class);
+		$message->expects($this->once())
+			->method('setTo')
+			->willReturn($message);
+		$message->expects($this->once())
+			->method('setSubject')
+			->willReturn($message);
+		$message->expects($this->once())
+			->method('setHtmlBody')
+			->willReturn($message);
+		$message->expects($this->once())
+			->method('setPlainBody')
+			->willReturn($message);
+		$message->expects($this->once())
+			->method('setFrom')
+			->willReturn($message);
+
+		$this->defaults->expects($this->exactly(2))
+			->method('getName')
+			->willReturn('ownCloud');
+
+		$this->mailer->expects($this->once())
+			->method('createMessage')
+			->willReturn($message);
+		$this->mailer->expects($this->once())
+			->method('send')
+			->with($message)
+			->willReturn([]);
+
+		$result = $this->createController()->resendToken('foo');
+		$this->assertEquals(
+			new Http\TemplateResponse(
+				'user_management', 'new_user/tokensendnotify',
+				[], 'guest'), $result);
+	}
+
+	public function providesResendTokenExceptionData() {
+		return [
+			['null_user'],
+			['email_not_set'],
+			['send_mail_fail']
+		];
+	}
+
+	/**
+	 * @dataProvider providesResendTokenExceptionData
```

Comment:
I think you should create different tests for each exception. You should use data provider to suply different data to the same test, but here you have different tests.

Reply:
Done as requested.